lib/graphql_devise/rails/routes.rb

```diff
@@ -1,111 +1,116 @@
+    def validate_operations!(param_operations)
+      GraphqlDevise::OperationChecker.call(
+        mutations: default_mutations,
+        queries:   default_queries,
+        **param_operations
       )
+    end
 
-      authenticatable_type = opts[:authenticatable_type] ||
-                             "Types::#{resource}Type".safe_constantize ||
-                             GraphqlDevise::Types::AuthenticatableType
+    def add_mutations!(prepared, additional)
+      all_mutations = prepared.merge(additional)
 
-      used_mutations = if only_operations.present?
-        default_mutations.slice(*only_operations)
-      else
-        default_mutations.except(*skipped_operations)
-      end
-      used_mutations.each do |action, mutation|
-        used_mutation = if custom_operations[action].present?
-          custom_operations[action]
-        else
-          new_mutation = Class.new(mutation)
-          new_mutation.graphql_name("#{resource}#{action.to_s.camelize(:upper)}")
-          new_mutation.field(:authenticatable, authenticatable_type, null: true)
-
-          new_mutation
-        end
-        used_mutation.instance_variable_set(:@resource_name, mapping_name)
-
-        GraphqlDevise::Types::MutationType.field("#{mapping_name}_#{action}", mutation: used_mutation)
-      end
-      additional_mutations.each do |action, mutation|
+      all_mutations.each do |action, mutation|
         GraphqlDevise::Types::MutationType.field(action, mutation: mutation)
       end
 
-      if (used_mutations.present? || additional_mutations.present?) &&
+      if all_mutations.present? &&
          (Gem::Version.new(GraphQL::VERSION) <= Gem::Version.new('1.10.0') || GraphqlDevise::Schema.mutation.nil?)
         GraphqlDevise::Schema.mutation(GraphqlDevise::Types::MutationType)
       end
+    end
 
-      used_queries = if only_operations.present?
-        default_queries.slice(*only_operations)
-      else
-        default_queries.except(*skipped_operations)
-      end
-      used_queries.each do |action, query|
-        used_query = if custom_operations[action].present?
-          custom_operations[action]
-        else
-          new_query = Class.new(query)
-          new_query.graphql_name("#{resource}#{action.to_s.camelize(:upper)}")
-          new_query.type(authenticatable_type, null: true)
-
-          new_query
-        end
-        used_query.instance_variable_set(:@resource_name, mapping_name)
-
-        GraphqlDevise::Types::QueryType.field("#{mapping_name}_#{action}", resolver: used_query)
-      end
-      additional_queries.each do |action, resolver|
+    def add_queries!(prepared, additional)
+      prepared.merge(additional).each do |action, resolver|
         GraphqlDevise::Types::QueryType.field(action, resolver: resolver)
       end
 
-      if (used_queries.blank? || additional_queries.present?) && GraphqlDevise::Types::QueryType.fields.blank?
```

Comment:
I think the names are better if you are more explicit about what the variable holds